### PR TITLE
[SPARK-20013][SQL]add a newTablePath parameter for renameTable in ExternalCatalog

### PR DESCRIPTION
**Pull request by windpiger**
## What changes were proposed in this pull request?

Currently when we create / rename a managed table, we should get the defaultTablePath for them in ExternalCatalog, then we have two defaultTablePath logic in its two subclass HiveExternalCatalog and InMemoryCatalog, additionally there is also a defaultTablePath in SessionCatalog, so till now we have three defaultTablePath in three classes. 
we'd better to unify them up to SessionCatalog

To unify them, we should move some logic from ExternalCatalog to SessionCatalog, renameTable is one of this.

while limit to the simple parameters in renameTable 
```
  def renameTable(db: String, oldName: String, newName: String): Unit
```
even if we move the defaultTablePath logic to SessionCatalog, we can not pass it to renameTable.

So we can add a newTablePath parameter  for renameTable in ExternalCatalog 
## How was this patch tested?
delete some tests in ExternalCatalogSuite which already existed in SessionCatalogSuite,
and move some other tests  in ExternalCatalogSuite which does not exist in SessionCatalogSuite

### CODE REVIEW

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogSuite.scala [hunk 1]
```diff
@@ -196,30 +196,6 @@ abstract class ExternalCatalogSuite extends SparkFunSuite with BeforeAndAfterEac
     catalog.dropTable("db2", "unknown_table", ignoreIfNotExists = true, purge = false)
   }
 
-  test("rename table") {
```
Comment (windpiger) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogSuite.scala]: already existed in [SessionCatalogSuite](https://github.com/apache/spark/blob/master/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalogSuite.scala#L361)

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogSuite.scala [hunk 2]
```diff
@@ -196,30 +196,6 @@ abstract class ExternalCatalogSuite extends SparkFunSuite with BeforeAndAfterEac
...
-    val catalog = newBasicCatalog()
-    assert(catalog.listTables("db2").toSet == Set("tbl1", "tbl2"))
-    catalog.renameTable("db2", "tbl1", "tblone")
-    assert(catalog.listTables("db2").toSet == Set("tblone", "tbl2"))
-  }
-
-  test("rename table when database/table does not exist") {
```
Comment (windpiger) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogSuite.scala]: already existed in [SessionCatalogSuite](https://github.com/apache/spark/blob/master/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalogSuite.scala#L393)

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogSuite.scala [hunk 3]
```diff
@@ -196,30 +196,6 @@ abstract class ExternalCatalogSuite extends SparkFunSuite with BeforeAndAfterEac
...
-    val catalog = newBasicCatalog()
-    intercept[AnalysisException] {
-      catalog.renameTable("unknown_db", "unknown_table", "unknown_table")
-    }
-    intercept[AnalysisException] {
-      catalog.renameTable("db2", "unknown_table", "unknown_table")
-    }
-  }
-
-  test("rename table when destination table already exists") {
```
Comment (windpiger) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogSuite.scala]: move to `SessionCatalogSuite`

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogSuite.scala [hunk 4]
```diff
@@ -732,40 +696,6 @@ abstract class ExternalCatalogSuite extends SparkFunSuite with BeforeAndAfterEac
     assert(!exists(db.locationUri))
   }
 
-  test("create/drop/rename table should create/delete/rename the directory") {
```
Comment (windpiger) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogSuite.scala]: move to `SessionCatalogSuite`

**Comment by windpiger** [comment]
cc @cloud-fan @gatorsmile

**Comment by gatorsmile** [comment]
Maybe we can close this one at first and revisit it later?

**Comment by gatorsmile** [comment]
We are closing it due to inactivity. please do reopen if you want to push it forward. Thanks!

**Pull request closed without merging**